pkg/binding/model.go

```diff
@@ -127,7 +140,19 @@ func newModel(annotationValue string) (*model, error) {
 		return nil, errors.New("sliceOfMaps elementType requires sourceKey and sourceValue to be present")
 	}
 
-	pathParts := strings.Split(path, ".")
+	reObf = regexp.MustCompile(`\[.*?\]`)
+	obfuscatedPath := reObf.ReplaceAllStringFunc(path, func(s string) string {
+		return "[" + strings.Repeat("_", len(s)-2) + "]"
+	})
+
+	obfPathParts := strings.Split(obfuscatedPath, ".")
+
+	pathParts := make([]string, len(obfPathParts))
+	start = 0
+	for i, part := range obfPathParts {
+		pathParts[i] = path[start : start+len(part)]
+		start += len(part) + 1
+	}
```

Comment:
given that json parser takes the full path as the argument, we do not need to split paths anymore, we should modify `model.path` field to be of type `string`.

Reply:
Yes, I explored this change, and that needs a lot of changes on the package. I will need some more time to work on this